File: C++/Step3/p1924.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
	int month,day;
	string days[] = {"SUN","MON","TUE","WED","THU","FRI","SAT"};
	int month_max_days[] = { 0,31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
	cin >> month >> day;
	int tmp = 0;
	for (int i = 0; i < month; i++)
		tmp += month_max_days[i];
	tmp += day;
	cout << days[tmp % 7];
}

File: JAVA/Step3/p1924.java
import java.util.Scanner;

public class Main{
  public static void main(String[] args) {
    Scanner input = new Scanner(System.in);
    int month = input.nextInt();
    int days = input.nextInt();
    int maxOfDays[] = {31,28,31,30,31,30,31,31,30,31,30,31};
    String daysOfTheweek[] = { "SUN", "MON", "TUE", "WED", "THU", "FRI", "SAT"};
    for(int i=0;i<month-1;i++) {
    	days+=maxOfDays[i];
    }
    System.out.println(daysOfTheweek[days%7]);
  }
}
